src/rpy_diffusive_flux.cc

```diff
@@ -0,0 +1,166 @@
+        auto flux_i = fluxes_(i)->view();
+        const double D = 1/(6*M_PI*viscosity_*a_i);
+
+        // fill flux with zeros initially
+        std::fill(flux_i.begin(), flux_i.end(), 0.);
+
+        // RPY flux from all species
+        for (const auto &j : state->getTypes())
+            {
+            const double a_j = 0.5 * diameters_(j);
+            auto rho_j = state->getField(j)->const_view();
+            auto mu_ex_j = (excess) ? excess->getDerivative(j)->const_view() : Field::ConstantView();
+            auto V_j = (external) ? external->getDerivative(j)->const_view() : Field::ConstantView();
+            for (int idx = 0; idx < mesh->shape(); ++idx)
+                {
+                bool Vidx_inf = std::isinf(V_j(idx));
+                if (V_j)
+                    {
+                    if (Vidx_inf)
+                        {
+                        if (V_j(idx) > 0 && rho_j(idx) > 0)
+                            {
+                            throw std::invalid_argument("Can't be V=+inf and have positive rho");
+                            }
+                        else if(V_j(idx) < 0)
+                            {
+                            throw std::invalid_argument("Can't have V=-inf ever, it is a mass sink");
+                            }
+                        }
+                    }
+                if ((Vidx_inf || std::isinf(V_j(idx-1))))
+                    {
+                    flux_i(idx) = 0.;
+                    continue;
+                    }
+
+                const auto x = mesh->lower_bound(idx);
+                
+                /* Considering symmetry about the center of the sphere and concentration gradient 
+                along the radial direction only, the flux at the center of the sphere is 0, i.e. 
+                density flux going into the center is equal to density flux out of the center. 
+                If this condition is not met, the symmetry assumption would be violated.*/
+                
+                if (x==0)
+                    {
+                    flux_i(idx) = 0.;
+                    continue;
+                    }
+                auto rho_x = mesh->interpolate(x, rho_i);
+                
+                //BD flux calculation
+                if(i==j)
+                    {
+                    auto dmu_ex = 0.0;
+                    if (mu_ex_j) dmu_ex += mesh->gradient(idx,mu_ex_j);
+                    if (V_j) dmu_ex += mesh->gradient(idx,V_j);
+                    flux_i(idx) += -D*(mesh->gradient(idx,rho_j) + rho_x*dmu_ex);
+                    }
+                
+                //RPY flux calculation   
+                const double d_ij = a_i + a_j;
+                
+                //To remove the concern about the lower bound value spill over the buffer sites
+                double lower_ig = ceil((x-d_ij-mesh->lower_bound())/mesh->step());
+                const int ig_low = mesh->bin(std::max(lower_ig, 0.));
```

Comment:
It is wrong to use bin on lower_ig, since lower_ig is an integer not a position.
```suggestion
                const int ig_low = std::max(ceil((x-d_ij-mesh->lower_bound())/mesh->step()), 0);
```